wandb/sdk/wandb_artifacts.py

```diff
@@ -469,10 +469,13 @@ def verify(self, root: Optional[str] = None):
     def save(self) -> None:
         if self._logged_artifact:
             return self._logged_artifact.save()
-
-        raise ValueError(
-            "Cannot call save on an artifact before it has been logged or in offline mode"
-        )
+        else:
+            if wandb.run is None:
+                run = wandb.init()
```

Comment:
context block would be cleaner here:

```
with wandb.init() as run:
   run.log_artifact(self)
```

Also, we may want a custom job_type here to easily filter these runs out.  Maybe `job_type="artifact"` or `job_type="auto"`...  We may also want this run to be in silent mode...

Reply:
3 great callouts (: Incoming shortly